### PR TITLE
Refactors Cyborg Module Rewriter tgui interface into TypeScript

### PR DESCRIPTION
**Pull request by mordent-goonstation**
## About the PR 

Converts Cyborg Module Rewriter code into TypeScript.

## Why's this needed? 

TypeScript refactor as initial phase of a "good practices" example. Not perfect or anything, but tries to show how to use assorted TypeScript features such as applying types to `data` and action payloads.

### CODE REVIEW

File: tgui/tsconfig.json [hunk 1]
```diff
@@ -19,7 +19,8 @@
     "module": "esnext",
     "moduleResolution": "node",
     "resolveJsonModule": true,
-    "isolatedModules": false
+    "isolatedModules": false,
+    "baseUrl": "packages"
```
Comment (mordent-goonstation) [inline tgui/tsconfig.json:23]: This change is to allow the aliased package (`common` is the one this PR uses, but would apply to others) be available to TypeScript. The `bem.js` file was converted to `bem.ts`.

I am not convinced this is the "right" way of doing this, will need to investigate further, so don't merge until it's been looked at a bit I guess.

@stylemistake @ZeWaka

**Review by mordent-goonstation — COMMENTED** [review]
lol, PR authors can't "Request changes" their own PR, so leaving this comment here instead I guess. I should probably go make this PR into a draft.

File: tgui/tsconfig.json (hunk 1 above)
Comment (stylemistake) [reply tgui/tsconfig.json:23]: /shrug, i think it worked without it.

Comment (mordent-goonstation) [reply tgui/tsconfig.json:23]: Possibly! I'll see if I would get the same warning in tgstation's repo. It certainly "works" in either case, just a few of the VS Code bits grump (Problems, and a warning).

**Review by ZeWaka — APPROVED** [review]
alreeady reviwed

**Pull request merged**